src/rust/lib/graph-editor/src/component/operator.rs

```diff
@@ -0,0 +1,271 @@
+//! component.
+//!
+//! This is also a step to avoid relying too much on the presence of the `Scene` within the UI
+//! component.
+use crate::prelude::*;
+
+use ensogl::display::DomScene;
+use ensogl::display::DomSymbol;
+use ensogl::display::Scene;
+use ensogl::display::Symbol;
+use ensogl::display::scene::View;
+use ensogl::display::traits::*;
+use ensogl::display;
+use ensogl::frp;
+use ensogl::gui::component::animation;
+
+
+
+// ==================================
+// === UI Component Helper Traits ===
+// ==================================
+
+/// Should be implemented by UI component that consist of `Symbol`. Provides access to the shapes
+/// and some helper methods for working with those shapes.
+pub trait NativeComponent {
+    /// Return all `Symbol`s that make up this component.
+    fn symbols(&self) -> Vec<SymbolType>;
+
+    /// Change the scene layer of all `Symbol`s.
+    fn set_layer(&self, layer:&View) {
+        self.symbols().iter().for_each(|symbol| {
+            match symbol{
+                SymbolType::Main(symbol)
+                | SymbolType::Visualisation(symbol)
+                => layer.add(symbol),
+            }
+        })
+    }
+
+    /// Remove the `Symbol`s from all scene layers.
+    fn unset_layers_all(&self, scene:&Scene) {
+        self.symbols().iter().for_each(|symbol|   match symbol{
+            SymbolType::Main(symbol)
+            | SymbolType::Visualisation(symbol)
+            =>scene.views.remove_symbol(symbol),
+        })
+    }
+}
+
+/// Should be implemented by UI component that consist of `DomSymbol`s. Provides access to the
+/// symbols some helper methods for working with them.
+pub trait HtmlComponent {
+    /// Return all `DomSymbol`s that make up this component.
+    fn elements(&self) -> Vec<DomSymbol>;
+
+    /// Change the `DomScene` of all `DomSymbol`s.
+    fn set_layer(&self, scene:&DomScene) {
+        self.elements().iter().for_each(|element| scene.manage(&element))
+    }
+}
+
+/// Provides information and functionality to resize an element. A complex UI component should
+/// implement this and propagate size and layout changes to all its sub-components.
+pub trait Resizable {
+    /// Set the size for the UI component.
+    fn set_size(&self, size:Vector3<f32>);
+    /// Return the size of the UI element.
+    fn size(&self) -> Vector3<f32>;
+}
+
+/// A component that owns a Network. Can be used to create external animations.
+pub trait Networked {
+    /// Return a reference to the components network.
+    fn network(&self) -> &frp::Network;
+}
+
+// ==================================
+// === Fullscreen Operator Handle ===
+// ==================================
+
+/// FullscreenOperatorCellHandle is a helper type that wraps a `FullscreenOperator` and applies an
+/// undos the operator on the inner component. This can be used to ensure that only a single
+/// component is made fullscreen at any time.
```

Comment:
If your docs refer to other entity in the code, it would be cool if the entity was defined earlier in the file so its easier to read the codebase without jumping the file up and down